content/articles/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/index.md

```diff
@@ -0,0 +1,222 @@
+
+```bash
+npm install -g typescript
+```
+
+The above command will install the Typescript compiler globally. This means any project you create on your computer can access Typescript dependencies without reinstalling the Typescript package when building another project.
+
+Run `tsc --version` to confirm if the compiler is installed.
+
+![TSC version checking](/engineering-education/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/tsc-version.png)
+
+> Note: Typescript code doesn't build directly on a browser (no browser will read Typescript directly). A browser will only read JavaScript code. To invoke any Typescript code, you need a compiler to transpile Typescript to browser executable JavaScript. Typescript will only help you in the development process so that you benefit from its features.
+
+### Creating Typescript config file
+
+To initialize the Typescript project with Node.js, create a [tsconfig.json](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html
+) file in the project root directory. The config file stores the options for the Typescript compiler. Alternatively, you can run `tsc --init` to automatically create this file in whatever directory you're in.
+
+![Generating tsconfig.json file](/engineering-education/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/generateing-tsconfig-file.png)
+
+This file is the Typescript configuration file. It specifies typescript compile options.
+
+![tsconfig.json file](/engineering-education/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/tsconfig-file.png)
+
+The `tsconfig.json` file has many options. It's good to know when to turn things on and off. TSC reads this file and uses these options to transpile Typescript into Brower readable JavaScript.
+
+Let's edit this config file to include the necessary option for running Typescript.
+
+- `"target": "es6"` - one key thing you need to define is the version of ECMAScript of JavaScript version that the compiler will output. If you need more advanced ECMAScript features, such as using the arrow function, `const`, and `let`, you can choose to use ES6 or a higher version. The target option allows you to use and follow the new JavaScript patterns. Most of today's browsers only understand the es6 version of JavaScript. we need to tell the Typescript compiler to transpile our Typescript code into es6 JavaScript.
+
+- `"module": "commonjs"` - this is the JavaScript module formatting system for structuring and organizing JavaScript code. This let's the compiler use module functions such as `require()` and objects such as `module.exports`.
+
+- `"rootDir": "./src"` - a directory where the input Typescript files are stored. I named the folder are `src`.
+
+- `"outDir": "./build"` - this an output directory where the output structure of the compiled JavaScript will be saved. In this case, JavaScript code will be saved in a folder called to `build` in the root directory of the main project.
```

Comment:
You have provided the `outDir` to be `build` here but you've used `dist` in the rest of the article. Please keep it consistent.